server/bin/pbench-verify-indexed.py

```diff
@@ -0,0 +1,187 @@
+
+import pbench
+
+
+_NAME_ = "pbench-reindex"
+
+tb_pat_r = (
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+tb_pat = re.compile(tb_pat_r)
+
+
+def gen_tb_list(archive):
+    """gen_tb_list - yield all controller/tarball names
+    """
+    with os.scandir(archive) as archive_scan:
+        # We are scanning the archive directory for all controller
+        # sub-directories.
+        for c_entry in archive_scan:
+            if c_entry.name.startswith(".") and c_entry.is_dir(follow_symlinks=False):
+                # Ignore the ".", "..", and any other ".*" subdirectories.
+                continue
+            if not c_entry.is_dir(follow_symlinks=False):
+                # NOTE: the pbench-audit-server should pick up and flag this
+                # unwanted condition.
+                continue
+            # We have a controller directory, now we scan the controller
+            # directory for all tar balls.
+            with os.scandir(c_entry.path) as controller_scan:
+                for entry in controller_scan:
+                    if not entry.is_file(follow_symlinks=False):
+                        # Tar balls can only be files.
+                        continue
+                    if entry.name.startswith("."):
+                        # Ignore any files starting with ".".
+                        continue
+                    match = tb_pat.fullmatch(entry.name)
+                    if not match:
+                        # Ignore any directory entries which do not match the
+                        # tar ball pattern.  Such entries should be flagged by
+                        # the server audit process.
+                        continue
+                    yield c_entry.name, entry.name
+
+
+def main(options):
+    if not options.cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable",
+            file=sys.stderr,
+        )
+        return 1
+
+    try:
+        config = pbench.PbenchConfig(options.cfg_name)
+    except pbench.BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 2
+
+    try:
+        archive_p = Path(config.ARCHIVE).resolve(strict=True)
+    except FileNotFoundError:
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE}, does not exist",
+            file=sys.stderr,
+        )
+        return 3
+
+    if not archive_p.is_dir():
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE}, is not a valid directory",
+            file=sys.stderr,
+        )
+        return 4
+
+    print("Verifying indexed state of tar balls")
+    start = pbench._time()
+
+    es = Elasticsearch(
+        [
+            {
+                "host": "elasticsearch.perf.lab.eng.bos.redhat.com",
+                "port": "9280",
+                "timeout": 600,
+            },
+        ]
+    )
```

Comment:
OK, it's hacked to a constant for a particular purpose; but as you've already got the config file parsed, which means you can't run this somewhere *without* a valid configuration file, I don't see any reason not to use the real configuration?